release-engineering/role-handbooks/release-image-promotion.md

```diff
@@ -1,46 +1,43 @@
+- Fork and clone the [`kubernetes/release` repository](https://github.com/kubernetes/release)
+- Fork the [`kubernetes/k8s.io` repository](https://github.com/kubernetes/k8s.io)
+- From the root of the `kubernetes/release` repository, run the following command to compile and install the release tools:
 
 ```shell
-go install ./cmd/cip-mm/... # Installs the manifest merge utility, cip-mm
+make release-tools
 ```
 
-- From the root of `k8s.io` repository
+- Run the following `krel` command to promote the images.  
+  - The `-i` flag runs the command in the interactive mode, so `krel` will ask you to confirm each step
+  - The `--fork` flag takes the GitHub username for a `kubernetes/k8s.io` fork that will be used to push the changes
+  - The `--tag` flag takes the version tag of the images that will be promoted
 
 ```shell
-cip-mm --base_dir=$GOPATH/src/k8s.io/k8s.io/k8s.gcr.io --staging_repo=gcr.io/k8s-staging-kubernetes --filter_tag=v1.19.0-rc.2
+krel promote-images -i --fork=<your-github-username> --tag=v1.20.0-beta.1
 ```
 
-When the release cut is not Alpha / Beta or RC we need to promote the next images as well, for example, if you are cutting `v1.19.0` you will need to promote the images for `1.19.1-rc.0` as well, so the commands will be something like:
+The following steps are taken by the `krel promote-images` command:
 
-```shell
-cip-mm --base_dir=$GOPATH/src/k8s.io/k8s.io/k8s.gcr.io --staging_repo=gcr.io/k8s-staging-kubernetes --filter_tag=v1.19.0
-
-cip-mm --base_dir=$GOPATH/src/k8s.io/k8s.io/k8s.gcr.io --staging_repo=gcr.io/k8s-staging-kubernetes --filter_tag=v1.19.1-rc.0
-```
+- Clone and update your `kubernetes/k8s.io` fork
+- Update the images manifest (`k8s.gcr.io/images/k8s-staging-kubernetes/images.yaml`) to add the appropriate image digests
+  - If you're cutting Alpha, Beta, or RC release, the image digests for the target release (e.g. `v1.20.0-beta.1`) will be added to the manifest
+  - If you're cutting a stable release, the image digests for the target release (e.g. `v1.20.0`) and the next RC release (e.g. `v1.20.1-rc.0`) will be added to the manifest
+- Create a branch and push it to your fork
+- Create a PR in the `kubernetes/k8s.io` repository with an explicit `/hold`
 
-_note_: If you clone the repo in another structure like `$GOPATH/src/github.com/kubernetes/k8s.io` you will nee to set the `--base_dir` as `--base_dir=$GOPATH/src/github.com/kubernetes/k8s.io/k8s.gcr.io`
-_note 2_: The `--filter_tag` you will add the release that you are working, like `v1.19.0-rc.3` / `v1.19.0-beta.1` / `v1.19.0`
+Example PRs:
 
-- After the command ran you should see updates to the `k8s.gcr.io/images/k8s-staging-kubernetes/images.yaml`
-- Create a branch and open a Pull Request with the new digests to promote. Add the links for cloudbuild jobs for `Mock Stage` and `Mock Release` and tag `@kubernetes/release-engineering` and add a explict `/hold` to be lifted before we start the `Official Release` step.
```

Comment:
> Add the links for cloudbuild jobs for `Mock Stage` and `Mock Release`

I'm not sure is it okay to drop this or not. We haven't been enforcing it, but should we start doing it? Do we need links to the logs in the image promotion PR?

Reply:
I think we should leave that in.
Ideally when trying to debug something in the future, you'll want some breadcrumbs to retrace your steps, so either:
- links to the GCB jobs OR
- a link to the release tracking issue (which should include links to the GCB jobs)

Reply:
@justaugustus Updated the PR to add a step for this.